framework/source/class/qx/lang/normalize/Array.js

```diff
@@ -251,6 +251,58 @@ qx.Bootstrap.define("qx.lang.normalize.Array", {
 
       return true;
     },
+    
+    /**
+     * The <code>find()</code> method returns a value in the array, if an element in the 
+     * array satisfies the provided testing function. Otherwise undefined is returned.
+     *
+     * <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find">MDN documentation</a> |
+     *
+     * @param callback {Function} Function to test for each element.
+     * @param obj {Object?} Value to use as <code>this</code> when executing <code>callback</code>.
+     * @return {Object} result, undefined if not found
+     */
+    find : function(callback, obj) {
+      if (qx.core.Environment.get("qx.debug")) {
+        qx.core.Assert.assertFunction(callback);
+      }
+      
+      var l = this.length;
+      for (var i = 0; i < l; i++) {
+        var value = this[i];
+        if (callback.call(obj || window, value, i, this)) {
```

Comment:
https://github.com/qooxdoo/qooxdoo/blob/master/tool/data/generator/oo.loader.tmpl.js#L3

window is defined for the oo variant, but it's defined only through closure, not really as a global variable. Sigh. We should probably fix oo.loader.tmpl.js to use an existing global context if one exists, e.g., in NodeJS, use *global*. I think what you have here is fine. We can assume that the window variable is global (even though in some environments, currently, it is not).